aws/aws-toolkit-vscode
# Tidy telemetry setup

## Description

This change tidies up some of the Telemetry related initialization code, to make it easier to maintain.

For the most part, existing logic was preserved. This change tried to keep functionality changes to a minimum. Here is what is noteworthy aside from the tidying up:
* the telemetry service now honors changes to the telemetry disabled/enabled configuration without requiring a VS Code restart
* toolkit no longer makes gratuitous write calls to update the telemetry configuration

## Testing

* launch toolkit, open toolkit log, change configuration values directly in JSON ("Enable", "Disable").  See that changes are reflected in toolkit log
* enable metrics, perform some activities, observe that metrics are sent
* disable metrics, perform some activities, observe (using breakpoints) that metrics are not sent  

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
